lib/Button/ButtonWithIcon.js

```diff
@@ -0,0 +1,27 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Button, Icon } from '../';
+
+const ButtonWithIcon = props => (
+  <div className="button-icon">
```

Comment:
If this is part of the button component then maybe something like `button__wrapper button__wrapper--icon` may keep us inline with our BEM patterns.